### ISSUE TITLE
RaggedTensor raises error with Keras model

### ISSUE THREAD

**Issue by kuan0808** [tensorflow/tensorflow#43591]
<em>Please make sure that this is a bug. As per our
[GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md),
we only address code/doc bugs, performance issues, feature requests and
build/installation issues on GitHub. tag:bug_template</em>

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow):yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04):Windows 10
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:
- TensorFlow installed from (source or binary):pip installed
- TensorFlow version (use command below):2.3.0
- Python version:3.7.0
- Bazel version (if compiling from source):
- GCC/Compiler version (if compiling from source):
- CUDA/cuDNN version:
- GPU model and memory:RTX 2060 6G/16G RAM

You can collect some of this information using our environment capture
[script](https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh)
You can also obtain the TensorFlow version with:
1. TF 1.0: `python -c "import tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"`
2. TF 2.0: `python -c "import tensorflow as tf; print(tf.version.GIT_VERSION, tf.version.VERSION)"`

**Describe the current behavior**
I builded a lstm model for one to one word classification:
```
model = Sequential()
model.add(layers.Input(shape=(None, 512), ragged=True))
model.add(layers.LSTM(32, return_sequences=True, dropout=0.4))
model.add(layers.TimeDistributed(layers.Dense(13, activation='softmax')))
model.compile(loss='categorical_crossentropy', optimizer='adam')
model.summary()
```
I then generated my x_train as a tf.RaggedTensor, with shape [10000, None, 512], dtype='float32'.
y_train as a tf.RaggedTensor, with shape [10000, None, 13], dtype='float32'.

Now if run it with:
`history = model.fit(x, y, epochs=10, verbose=1)`

I'd get:
> TypeError: Failed to convert object of type <class 'tensorflow.python.ops.ragged.ragged_tensor.RaggedTensor'> to Tensor. Contents: tf.RaggedTensor(values=Tensor("sequential_42/time_distributed_46/dense_54/Softmax:0", shape=(None, 13), dtype=float32), row_splits=Tensor("sequential_42/time_distributed_46/RaggedFromRowLengths/control_dependency:0", shape=(None,), dtype=int64)). Consider casting elements to a supported type.
**Describe the expected behavior**
TimeDistributed layer should support RaggedTensor if the underlying model can process slice of RaggedTensor and output dimension is matched.
**Standalone code to reproduce the issue**
```
import tensorflow as tf
from tensorflow.keras import layers, Model, Sequential
import numpy as np

x = tf.RaggedTensor.from_row_splits(np.ones((100, 512)), [0, 4, 20, 100])
y = tf.RaggedTensor.from_row_splits(np.ones((100, 13)), [0, 4, 20, 100])
print(x.shape)
print(y.shape)
model = Sequential()
model.add(layers.Input(shape=(None, 512), ragged=True))
model.add(layers.LSTM(32, return_sequences=True, dropout=0.4))
model.add(layers.TimeDistributed(layers.Dense(13, activation='softmax')))
model.compile(loss='categorical_crossentropy', optimizer='adam')
model.summary()
history = model.fit(x=x, y=y, epochs=10, verbose=1)
```

**Comment by amahendrakar** [comment]
Was able to reproduce the issue with TF v2.3 and TF-nightly. Please find the gist of it [here](https://colab.research.google.com/gist/amahendrakar/c3822d0a920d389d8a5ca6312bc1e424/43591.ipynb#scrollTo=DDbwlfhWR0vu). Thanks!

**Comment by jvishnuvardhan** [comment]
@kuan0808 I guess the root-cause is the mismatch in input arguments (x and y) to `model.fit`. Please [check](https://www.tensorflow.org/api_docs/python/tf/keras/Model#fit) what is supported by `model.fit`.

The error clearly says 

> TypeError: Failed to convert object of type <class 'tensorflow.python.ops.ragged.ragged_tensor.RaggedTensor'> to Tensor. Contents: tf.RaggedTensor(values=Tensor("sequential/time_distributed/dense/Softmax:0", shape=(None, 13), dtype=float32), row_splits=Tensor("sequential/time_distributed/RaggedFromRowLengths/control_dependency:0", shape=(None,), dtype=int64)). Consider casting elements to a supported type.

Can you please use supported types and let us know how it progresses. Thanks!

**Comment by kuan0808** [comment]
Do you mean the dimension mismatch?
I have check the dimension, and they should be matched. And both x, y are ragged tensors, which is supposed to supported by model.fit I think🤔

**Comment by foxik** [comment]
Note that supporting RaggedTensors as targets (e.g., in `model.fit`) is a frequent request, see #44988, #44112, #43591, #43093, #42320, #41810.

Some partial progress is in pull requests #45060 and #45015, which will allow using custom losses in Keras model (but existing losses like MSE or (S)CE will still not work).

**Comment by foxik** [comment]
Note that I created a new feature request #45403 to support RaggedTensors in standard Keras loss functions.

**Comment by rmothukuru** [comment]
@kuan0808,
Issue has been resolved in **`TF Nightly`** (`2.5.0-dev20210302`). Please find the [Gist of the working code](https://colab.research.google.com/gist/rmothukuru/9d24fde1d43daf19d376723196d97425/43591.ipynb#scrollTo=c9xLppX3TOis). Thanks!
